pygmt/tests/test_clib_to_numpy.py

```diff
@@ -0,0 +1,140 @@
+"""
+Tests for the _to_numpy function in the clib.conversion module.
+"""
+
+import numpy as np
+import numpy.testing as npt
+import pandas as pd
+import pytest
+from pygmt.clib.conversion import _to_numpy
+
+
+def _check_result(result, expected_dtype):
+    """
+    A helper function to check if the result of the _to_numpy function is a C-contiguous
+    NumPy array with the expected dtype.
+    """
+    assert isinstance(result, np.ndarray)
+    assert result.flags.c_contiguous
+    assert result.dtype.type == expected_dtype
+
+
+########################################################################################
+# Test the _to_numpy function with Python built-in types.
+########################################################################################
+@pytest.mark.parametrize(
+    ("data", "expected_dtype"),
+    [
+        pytest.param([1, 2, 3], np.int64, id="int"),
+        pytest.param([1.0, 2.0, 3.0], np.float64, id="float"),
```

Comment:
Not sure if it's necessary, but Python does have built-in [complex](https://docs.python.org/3/library/functions.html#complex) number types (https://docs.python.org/3/library/stdtypes.html#numeric-types-int-float-complex)

```suggestion
        pytest.param([1.0, 2.0, 3.0], np.float64, id="float"),
        pytest.param(
            [complex(+1), complex(-2j), complex("-Infinity+NaNj")],
            np.complex128,
            id="complex",
        ),
```

The Python standard library also includes [fractions.Fraction](https://docs.python.org/3/library/fractions.html#fractions.Fraction) and [decimal.Decimal](https://docs.python.org/3/library/decimal.html#decimal.Decimal), but I don't know if anyone really uses those.

Reply:
> The Python standard library also includes [fractions.Fraction](https://docs.python.org/3/library/fractions.html#fractions.Fraction) and [decimal.Decimal](https://docs.python.org/3/library/decimal.html#decimal.Decimal), but I don't know if anyone really uses those.

They can't be converted to a numpy array, so PyGMT can't support it anyway.